client/reducers/events.js

```diff
@@ -116,14 +116,39 @@ const eventsReducer = createReducer(initialState, {
             selectedEvent: payload,
         }
     ),
-    [EVENTS.ACTIONS.OPEN_EVENT_DETAILS]: (state, payload) => (
-        {
-            ...state,
-            showEventDetails: payload,
-            highlightedEvent: payload,
-            readOnly: false,
+    [EVENTS.ACTIONS.OPEN_EVENT_DETAILS]: (state, payload) => {
+        if (payload.lock_user) {
+            let _events = cloneDeep(state.events)
+            let event = payload
+
+            // Change dates to moment objects
```

Comment:
we are already doing that in more and more different places. It starts to alert me.
Just to know how we could factorize it later, do you remember in which scenario the dates here are not moment objects ?